legacy/firmware/layout2.c

```diff
@@ -653,6 +653,50 @@ void layoutPublicKey(const uint8_t *pubkey) {
                     str[1], str[2], str[3], NULL);
 }
 
+void layoutXPUB(const char *xpub, int index, int page, bool ours) {
+  if (layoutLast != layoutAddress && layoutLast != layoutXPUB) {
+    layoutSwipe();
+  } else {
+    oledClear();
+  }
+  layoutLast = layoutXPUB;
+  char desc[] = "XPUB #__ _/2 (______)";
+  if (index + 1 >= 10) {
+    desc[6] = '0' + (((index + 1) / 10) % 10);
+    desc[7] = '0' + ((index + 1) % 10);
+  } else {
+    desc[6] = '0' + ((index + 1) % 10);
+    desc[7] = ' ';
+  }
+  desc[9] = '1' + page;
+  if (ours) {
+    desc[14] = 'y';
+    desc[15] = 'o';
+    desc[16] = 'u';
+    desc[17] = 'r';
+    desc[18] = 's';
+    desc[19] = ')';
+    desc[20] = 0;
+  } else {
+    desc[14] = 'o';
+    desc[15] = 't';
+    desc[16] = 'h';
+    desc[17] = 'e';
+    desc[18] = 'r';
+    desc[19] = 's';
+  }
+
+  xpub += page * 81;  // skip 81 characters per page
```

Comment:
maybe add an explanation, like `21 chars per line * 4 lines, minus 3 for "..."`? :)